src/Authenticator/HttpDigestAuthenticator.php

```diff
@@ -78,33 +78,31 @@ public function __construct(IdentifierInterface $identifier, array $config = [])
     public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
     {
         $digest = $this->_getDigest($request);
-        if (empty($digest)) {
+        if ($digest === null) {
             return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
         }
 
-        $user = $this->_identifier->identify([
-            IdentifierInterface::CREDENTIAL_USERNAME => $digest['username']
-        ]);
+        $identity = $this->_identifier->identify([IdentifierInterface::CREDENTIAL_USERNAME => $digest['username']]);
 
-        if (empty($user)) {
+        if (empty($identity)) {
             return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND);
         }
 
         if (!$this->validNonce($digest['nonce'])) {
             return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
         }
 
-        $field = $this->_config['fields'][IdentifierInterface::CREDENTIAL_PASSWORD];
-        $password = $user[$field];
+        $passwordField = $this->getConfig('fields')[IdentifierInterface::CREDENTIAL_PASSWORD];
```

Comment:
How is that better than `$this->_config['fields'][IdentifierInterface::CREDENTIAL_PASSWORD]` ?

Reply:
calling methods is always better. You can set the instance config trait to throw exceptions in case the keys do not exist, whereas the array access will simply throw warnings/notices, right?